cpp/cmake/Modules/ConfigureArrow.cmake

```diff
@@ -1,94 +1,55 @@
-unset(PARALLEL_BUILD)            
-if($ENV{TRAVIS})
-    if(NOT DEFINED ENV{CMAKE_BUILD_PARALLEL_LEVEL})
-        message(STATUS "Disabling Parallel CMake build on Travis")
-    else()
-        set(PARALLEL_BUILD --parallel)
-        message(STATUS "Using $ENV{CMAKE_BUILD_PARALLEL_LEVEL} build jobs on Travis")
-    endif(NOT DEFINED ENV{CMAKE_BUILD_PARALLEL_LEVEL})
-else()
-    set(PARALLEL_BUILD --parallel)
-    message("STATUS Enabling Parallel CMake build")
-endif($ENV{TRAVIS})
+if(result)
+    message(FATAL_ERROR "CMake step for Arrow failed: ${result}")
+endif()
 
 execute_process(
-    COMMAND ${CMAKE_COMMAND} --build ${PARALLEL_BUILD} ..
-    RESULT_VARIABLE ARROW_BUILD
-    WORKING_DIRECTORY ${ARROW_ROOT}/build)
+    COMMAND ${CMAKE_COMMAND} --build .
+    RESULT_VARIABLE result
+    WORKING_DIRECTORY ${CMAKE_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/thirdparty/arrow-download)
 
-if(ARROW_BUILD)
-    message(FATAL_ERROR "Building Arrow failed: " ${ARROW_BUILD})
-endif(ARROW_BUILD)
+if(result)
+    message(FATAL_ERROR "Build step for Arrow failed: ${result}")
+endif()
 
-set(ARROW_GENERATED_IPC_DIR 
-    "${ARROW_ROOT}/build/src/arrow/ipc")
+# Add transitive dependency: Flatbuffers
+set(FLATBUFFERS_ROOT ${CMAKE_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/thirdparty/arrow-build/flatbuffers_ep-prefix/src/flatbuffers_ep-install/)
+include_directories(${FLATBUFFERS_ROOT}/include/)
+link_directories(${FLATBUFFERS_ROOT}/lib/)
 
+set(ARROW_GENERATED_IPC_DIR ${CMAKE_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/thirdparty/arrow-build/src/arrow/ipc/)
 configure_file(${ARROW_GENERATED_IPC_DIR}/File_generated.h ${CMAKE_SOURCE_DIR}/include/cudf/ipc_generated/File_generated.h COPYONLY)
 configure_file(${ARROW_GENERATED_IPC_DIR}/Message_generated.h ${CMAKE_SOURCE_DIR}/include/cudf/ipc_generated/Message_generated.h COPYONLY)
 configure_file(${ARROW_GENERATED_IPC_DIR}/Schema_generated.h ${CMAKE_SOURCE_DIR}/include/cudf/ipc_generated/Schema_generated.h COPYONLY)
 configure_file(${ARROW_GENERATED_IPC_DIR}/Tensor_generated.h ${CMAKE_SOURCE_DIR}/include/cudf/ipc_generated/Tensor_generated.h COPYONLY)
 
-message(STATUS "Arrow installed here: " ${ARROW_ROOT}/install)
-set(ARROW_LIBRARY_DIR "${ARROW_ROOT}/install/lib")
-set(ARROW_INCLUDE_DIR "${ARROW_ROOT}/install/include")
-
-find_library(ARROW_LIB arrow
-             NO_DEFAULT_PATH
-             HINTS "${ARROW_LIBRARY_DIR}")
-
-if(ARROW_LIB)
-    message(STATUS "Arrow library: " ${ARROW_LIB})
-    set(ARROW_FOUND TRUE)
-endif(ARROW_LIB)
-
-set(FLATBUFFERS_ROOT "${ARROW_ROOT}/build/flatbuffers_ep-prefix/src/flatbuffers_ep-install")
-
-message(STATUS "FlatBuffers installed here: " ${FLATBUFFERS_ROOT})
-set(FLATBUFFERS_INCLUDE_DIR "${FLATBUFFERS_ROOT}/include")
-set(FLATBUFFERS_LIBRARY_DIR "${FLATBUFFERS_ROOT}/lib")
-
-add_definitions(-DARROW_METADATA_V4)
-add_definitions(-DARROW_VERSION=1000)
-
-
-
-
-
-
+set(ENV{ARROW_HOME} ${CMAKE_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/thirdparty/arrow-install/)
```

Comment:
Please revert these changes. The point is to have arrow in an obvious location with settings that are clear.

Reply:
ok!

Reply:
@mt-jones Like this?
https://github.com/BlazingDB/cudf/blob/upgrade-arrow/cpp/cmake/Modules/ConfigureArrow.cmake#L98